src/runtime/pkg/katautils/config.go

```diff
@@ -1232,6 +1247,70 @@ func decodeConfig(configPath string) (tomlConfig, string, error) {
 	return tomlConf, resolved, nil
 }
 
+func decodeDropIns(mainConfigPath string, tomlConf *tomlConfig) error {
+	configDir := filepath.Dir(mainConfigPath)
+	dropInDir := filepath.Join(configDir, "config.d")
+
+	files, err := ioutil.ReadDir(dropInDir)
```

Comment:
In the documentation and discussion, you stated several times that drop-in files would be read in "alphabetical order". This code uses `ReadDir`, but does not sort the output. The fact that `readdir` does not guarantee order is discussed here https://stackoverflow.com/questions/8977441/does-readdir-guarantee-an-order. Here is an example of `ReadDir` output not being in alphabetical order in Go. https://thedeveloperblog.com/go/readdir-go.

So I think that you probably want to add a sort here to ensure the kind of stable semantics we want.

Reply:
@c3d Thanks for your review!  Well, I'm not quite convinced we have a problem here since the former link pertains to POSIX `readdir()` which doesn't sort the output but is not the function called here, and the latter one pertains to [Readdir() in os.File](https://pkg.go.dev/os#File.Readdir) which also doesn't sort the output but, again, is not the function called here either.  What's called here is [ioutil.ReadDir()](https://pkg.go.dev/io/ioutil#ReadDir) which is documented to sort the output by filename.

Now, it's true that the docs don't specify "alphabetical order" explicitly so you're right, I might have assumed more than I was afforded by the docs. I'd say that it's a safe assumption though, but if there's a consensus among reviewers that we should sort as well this should be easy to add.